src/app/ui/passworddialog.cpp

```diff
@@ -0,0 +1,24 @@
+#include "passworddialog.h"
+#include "ui_passworddialog.h"
+
+PasswordDialog::PasswordDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::PasswordDialog)
+{
+    ui->setupUi(this);
+}
+
+PasswordDialog::~PasswordDialog()
+{
+    delete ui;
+}
+
+void PasswordDialog::accept()
+{
+    PasswordDialog::~PasswordDialog();
```

Comment:
u don`t need to remove password dialog
You should read documentation of QDialog
" This includes deleting the dialog while it is visible."
You should in ui connect ok clicked to QDialog accepted.